src/QLContext.actor.cpp

```diff
@@ -376,12 +375,13 @@ struct CompoundIndexPlugin : IndexPlugin, ReferenceCounted<CompoundIndexPlugin>,
 				// for all new entries going to be written, before we clear the potentially existing old index entries,
 				// we need to make sure there is no existing unique index for that new value under path
 				// dbName+collectionName+"metadata"+"indices"+encodedIndexName+encodedValue
-				if (self->flowControlLock.present()) {
-					// When building the unique index, each record out of the table scan needs to wait for the previous
-					// one finished duplication detecting and index record writing. And thus the following section
-					// before the `lock.release()` call, needs to be protected using a mutex lock.
-					Void _ = wait(self->flowControlLock.get()->lock->take(1));
-				}
+
+				// When building the unique index, each record out of the table scan needs to wait for the previous
+				// one finished duplication detecting and index record writing. And thus the following section
+				// before the `lock.release()` call, needs to be protected using a mutex lock.
+				state FlowLock::Releaser releaser(*self->flowControlLock.get(), 1);
+				Void _ = wait(self->flowControlLock.get()->take(1));
```

Comment:
It's still `Optional`, why we are not checking for `present()`?

Reply:
Oh the `ASSERT` was lost through copy and paste. Let me add it back.

Reply:
Ah! its because unique index precondition.

Reply:
Yeah the unique flag should have ensured it, but will add an ASSERT just to be safe.